react-ui/src/App.jsx

```diff
@@ -59,30 +40,57 @@ function App() {
           <div className="app">
             <BrowserRouter>
               <AppHelmet />
-              <Routes>
-                <Route exact path="/" element={<Navigate replace to="home" />} />
-                <Route exact path="/authorizeCallback" element={<AuthorizeCallback />} />
-                <Route exact path="/home" element={<Home />} />
-                <Route exact path="/contests" element={<Contests />} />
-                <Route exact path="/contests/:contestId" element={<Contest />} />
-                <Route exact path="/contests/:contestId/entry/:entryId" element={<Entry />} />
-                <Route exact path="/mod/review" element={<ReviewSubmissions />} />
-                <Route exact path="/submission" element={<Submission />} />
-                <Route
-                  exact
-                  path="/submit"
-                  element={<Navigate replace state={{ defaultTab: 1 }} to="/submission" />}
-                />
-                <Route exact path="/hallOfFame" element={<HallOfFame />} />
-                <Route exact path="/profile/settings" element={<Settings />} />
-              </Routes>
+              <ModalSwitch />
             </BrowserRouter>
             <CustomSnackbar />
           </div>
         </SWRConfig>
-      </ThemeProvider>
+      </CustomThemeProvider>
     </>
   );
 }
 
+function ModalSwitch() {
+  const location = useLocation();
+
+  const background = location.state && location.state.background;
```

Comment:
I recently enabled whichever ES version has optional chaining, so this can just be `const background = location.state?.background;`.

Reply:
Perfect, that's much cleaner